bigquery_etl/query_scheduling/task.py

```diff
@@ -205,6 +231,13 @@ def with_dependencies(self, client, dag_collection):
             upstream_task = dag_collection.task_for_table(table[0], table[1])
 
             if upstream_task is not None:
-                dependencies.append(upstream_task)
+                # ensure there are no duplicate dependencies
+                # manual dependency definitions overwrite automatically detected ones
+                if not any(
+                    d.dag_name == upstream_task.dag_name
+                    and d.task_id == upstream_task.task_name
+                    for d in self.depends_on
+                ):
+                    dependencies.append(upstream_task)
```

Comment:
Handle duplicate task dependencies listed between the `depends_on` list and automatically inferred dependencies here